### PR TITLE
Make Button respect accessibilityRole passed in props

### PR DESCRIPTION
**Pull request by nakambo**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

### Description of changes

e.g. I'd like to be able to use things like menuitem or similar, which indicates a button-like control in a list of similar controls.

### Verification

* I see the expected role now with tools, e.g. Accessibility Insights.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by nakambo** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: packages/components/Button/src/useButton.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const useButton = (props: ButtonProps): ButtonInfo => {
        */
       ...(isDisabled && { disabled: isDisabled }),
       accessible: true,
-      accessibilityRole: 'button',
+      accessibilityRole: accessibilityRole || 'button',
```
Comment (Saadnajmi) [inline packages/components/Button/src/useButton.ts:27]: Couldn't this also be achieved by doing a `...rest` spread at the bottom of this hook (after line 34)?

Comment (nakambo) [reply packages/components/Button/src/useButton.ts:27]: `rest` is used by `usePressableState`, and the result is mixed in _before_ the overrides here, so it looks like these could be intentional. I do see that certain other properties (`enableFocusRing`, `iconPosition`) do apply the same pattern that I am proposing to apply here.

**Comment by rurikoaraki** [comment]
We do have a Menu control now, FWIW. Change seems reasonable to me though

**Review by rurikoaraki — APPROVED** [review]

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**